packages/runtime/container-runtime/src/opLifecycle/opGroupingManager.ts

```diff
@@ -22,10 +22,12 @@ interface IGroupedMessage {
 	compression?: string;
 }
 
-function isGroupContents(
-	opContents: IGroupedBatchMessageContents | { type?: unknown } | undefined,
-): opContents is IGroupedBatchMessageContents {
-	return opContents?.type === OpGroupingManager.groupedBatchOp;
+function isGroupContents(opContents: unknown): opContents is IGroupedBatchMessageContents {
+	return (
+		typeof opContents === "object" &&
+		// eslint-disable-next-line @typescript-eslint/dot-notation
```

Comment:
Why not `opContents?.type` like it was / like the linter wants?  What's the difference?

Reply:
Using `?.type`: `Property 'type' does not exist on type 'object'.ts(2339)`
While I can't find any documentation to back it up, my understanding is that bracket notation is has less checking. Even though a literal is used typescript doesn't do a full type check. I would guess that is one of the reasons for the eslint rule, but I also didn't dig deep to find its origin. For javascript everything says the two syntaxes are runtime equivalent the only difference being that bracket notation allows access to more properties - properties that can't be expressed through dot notation.